BlueprintUICommonControls/Sources/AccessibilityFocus.swift

```diff
@@ -0,0 +1,117 @@
+import BlueprintUI
+import UIKit
+
+/// Enables VoiceOver focus to jump to the wrapped element
+/// either on appearance or via a trigger that can be manually fired.
+public struct AccessibilityFocus: Element {
+
+    /// The element that will have the focus.
+    public var wrapping: Element
+
+    /// The type of accessibility notification that will be triggered.
+    public var notificationType: UIAccessibility.Notification
+
+    /// A object that can be held on to by the caller to manually trigger a focus.
+    public var trigger: Trigger
+
+    // MARK: Element
+
+    public var content: ElementContent {
+        ElementContent(child: wrapping)
+    }
+
+    public func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
+        View.describe { config in
+            config[\.isAccessibilityElement] = false
+            config.apply { view in
+                view.apply(model: self)
+            }
+        }
+    }
+}
+
+// MARK: Backing view
+
+extension AccessibilityFocus {
+    private final class View: UIView {
+
+        func apply(model: AccessibilityFocus) {
+            model.trigger.action = { [weak self] in
+                guard let self = self else { return }
+                UIAccessibility.post(notification: model.notificationType, argument: self)
+            }
+
+            /// We need to wait a moment before firing the focus, to ensure that
+            /// we are not competing with UIAccessibility for the focus.
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
+                model.trigger.fireImmediately()
+            }
+        }
+    }
+}
+
+// MARK: Trigger
+
+extension AccessibilityFocus {
+
+    /// A trigger that can be used to manually fire an accessibility focus.
+    public class Trigger {
```

Comment:
Just a thought, perhaps we should add an optional identifier to the trigger to help with managing a bunch of them